### ISSUE TITLE
ArcFaceLoss num_classes and embedding_size

### ISSUE THREAD

**Issue by brianw0924** [KevinMusgrave/pytorch-metric-learning#597]
Sorry I don't really understand how arcface work

if embedding size is 512, and I have 214 class in training dataset

1. My model final layer should output 512, why not 214?
2. Did arcface calculate cross entropy loss? if yes, why the model's final layer is not linear(512, 214)? (where's the weight that project 512 to 214 then calculate cross entropy loss?)

Is it because in arcface, the weight is normalized to 1, so your package automatically include the final layer transform?

Thank you

**Comment by KevinMusgrave** [comment]
ArcFacLoss contains the weights that project from 512 to 214. That's why it needs to know `embedding_size=512, num_classes=214`.
